source/renderer/app/features/menu-updater/makeReactionCallback.js

```diff
@@ -0,0 +1,43 @@
+// @flow
+import { matchPath } from 'react-router-dom';
+import { WalletSettingsStateEnum } from '../../../../common/ipc/api';
+import { ROUTES } from '../../routes-config';
+import type { MakeReactionCallbackArgs } from './types';
+
+const walletRoutes = Object.values(ROUTES.WALLETS);
+
+const makeReactionCallback = ({
+  stores: { app, profile, router, staking, uiDialogs },
+  rebuildApplicationMenu,
+}: MakeReactionCallbackArgs) => () => {
+  let walletSettingsState = WalletSettingsStateEnum.hidden;
```

Comment:
I'm not really sure how reaction works but another solution is to use `useEffect`. e.g:

```js
const ListenMenuChanges = observer(() => {

useEffect(() => {
 /* logic goes here ... */

rebuildApplicationMenu.send({
    isNavigationEnabled: profile.areTermsOfUseAccepted,
    walletSettingsState,
  });
},[])

return null
})
```

And use this on the app:

```
<MenuUpdaterProvider>
<App />
<ListenMenuChanges />
</MenuUpdaterProvider>
```

Reply:
The `Reaction` class provides the api around the [mobx autorun feature](https://mobx.js.org/reactions.html) which is a way of making side effects on changes to the store so my case was a perfect use case for it. However you make a good point, using `useEffect` probably would simplify things in this case. Summarising: I don't have any feature to pass down the react tree through context and don't have to use the `autorun` and `Reaction` necessarily, so there is no point in calling this a Feature. I will use the `useEffect` instead of `Reaction` and move the implementation out of the features directory then.